src/core/extension.ts

```diff
@@ -289,54 +257,55 @@ export class BallerinaExtension {
         return 0;
     }
 
-    checkCompatibleVersion(pluginVersion: string, ballerinaVersion: string): void {
-        if (pluginVersion === '2.0.0' && ballerinaVersion.match(SWAN_LAKE_REGEX)) {
-            return;
-        }
-        const versionCheck = this.compareVersions(pluginVersion, ballerinaVersion);
-
-        if (versionCheck > 0) {
-            // Plugin version is greater
-            this.showMessageOldBallerina();
-            this.telemetryReporter.sendTelemetryEvent(TM_EVENT_OLD_BAL_HOME);
-            return;
+    async getBallerinaVersion(ballerinaHome: string, overrideBallerinaHome: boolean): Promise<string> {
+        // if ballerina home is overridden, use ballerina cmd inside distribution
+        // otherwise use wrapper command
+        let distPath = "";
+        if (overrideBallerinaHome) {
+            distPath = path.join(ballerinaHome, "bin") + path.sep;
         }
-
-        if (versionCheck < 0) {
-            // Ballerina version is greater
-            this.showMessageOldPlugin();
-            this.telemetryReporter.sendTelemetryEvent(TM_EVENT_OLD_BAL_PLUGIN);
+        let exeExtension = "";
+        if (process.platform === 'win32') {
+            exeExtension = ".bat";
         }
-    }
 
-    getBallerinaVersion(ballerinaHome: string, overrideBallerinaHome: boolean): Promise<string> {
-        if (overrideBallerinaHome && !ballerinaHome) {
-            throw new AssertionError({
-                message: "Trying to get ballerina version without setting ballerina home."
-            });
-        }
-        // if ballerina home is overridden, use ballerina cmd inside distribution
-        // otherwise use wrapper command
-        const balCmd = this.getBallerinaCmd(overrideBallerinaHome ? ballerinaHome : "");
-        return new Promise((resolve, reject) => {
-            exec(balCmd + ' version', (err, stdout, stderr) => {
+        let ballerinaExecutor = '';
+        const balPromise: Promise<string> = new Promise((resolve, reject) => {
+            exec(distPath + 'bal' + exeExtension + ' version', (_err, stdout, stderr) => {
```

Comment:
why we are dropping central method `getBallerinaCmd` ? if we have such method we don't need to check `process.platform === 'win32'` in other places

Reply:
We can make promise based resolution for `ballerina` and `bal` inside `getBallerinaCmd`. Also with these changes are we also dropping support `overrideBallerinaHome` config feature ?

Reply:
We removed the `getBallerinaCmd()` because now we do a one-time initialization to the `ballerinaCmd` variable of the ballerinaExtension instance while initializing the extension (Now that we need the cmd to get the version, promise based resolution to the cmd is done while retrieving the ballerina version). We only use these platform-specific comparisons when initializing the `ballerinaCmd` variable. Afterward, we directly refer to the `ballerinaCmd` value everywhere else, without re-evaluating the value.

Reply:
That make sense. Can other modules refer this command as `ballerinaExtension.ballerinaCmd` ?

Reply:
Yes, all the ballerinaExtension references, which are used in LS, debugger, and commands, can refer to the ballerinaCmd value.